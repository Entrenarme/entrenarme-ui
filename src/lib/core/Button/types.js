export type ButtonType =
  | 'primary'
  | 'secondary'
  | 'warning'
  | 'link'
  | 'error'
  | 'success'
  | 'add';

export type ButtonState = 'outline' | 'disabled' | 'loading' | 'icon';
