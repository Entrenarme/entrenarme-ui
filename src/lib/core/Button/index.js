// @flow
import * as React from 'react';
import MButton from '@material-ui/core/Button';
import CircularProgress from '@material-ui/core/CircularProgress';
import styled from 'styled-components';

import * as fonts from '../../helpers/fonts';
import colors from '../../helpers/colors';

type ButtonProps = {
  options: {
    type: ButtonType,
    state: ButtonState,
    colorVariant: ButtonColorVariant,
    disabled: ButtonDisabled,
  },
};

const getBgColor = (
  buttonType: ButtonType,
  options: 'default' | 'active',
  state: ButtonState,
  disabled: ButtonDisabled,
  colorVariant: ButtonColorVariant,
) => {
  if (buttonType === 'link' || buttonType === 'outline') {
    return colors.white.default;
  }
  if (buttonType === 'add' || disabled) {
    return colors.gray.default;
  }
  if (state === 'error' || state === 'success') {
    return colors[state].default;
  }
  return colors[colorVariant || state].default;
};

const getColor = (
  buttonType: ButtonType,
  state: ButtonState,
  colorVariant: ButtonColorVariant,
  disabled: ButtonDisabled,
) => {
  if (buttonType === 'outline') {
    return colors[colorVariant].default;
  }
  if (disabled) {
    return colors.gray.dark;
  }
  if (state === 'loading') {
    if (buttonType === 'link') {
      return colors.gray.loading;
    }
    if (buttonType === 'add') {
      return colors.secondary.loading;
    }
    return colors.white.loading;
  }
  if (buttonType === 'link' || buttonType === 'add') {
    return colors.secondary.default;
  }
  return colors.white.default;
};

const SButton = styled(MButton)`
  background-color: ${(props: ButtonProps) =>
    getBgColor(
      props.options.type,
      'default',
      props.options.state,
      props.options.disabled,
      props.options.colorVariant,
    )} !important;
  ${(props: ButtonProps) =>
    props.options.type === 'outline'
      ? `	border: 2px solid ${getColor(
          props.options.type,
          props.options.state,
          props.options.colorVariant,
          props.options.disabled,
        )} !important`
      : props.options.type === 'add'
        ? `border: 2px dashed ${colors.gray.dark} !important;`
        : ''};
  :hover {
    background-color: ${(props: ButtonProps) =>
      getBgColor(
        props.options.type,
        'active',
        props.options.state,
        props.options.disabled,
        props.options.colorVariant,
      )} !important;
  }
  span {
    color: ${(props: ButtonProps) =>
      getColor(
        props.options.type,
        props.options.state,
        props.options.colorVariant,
        props.options.disabled,
      )};
    font-family: ${fonts.mainFont};
  }
`;

type Props = {
  /** This is what is gonig to be rendered inside the button */
  children: React.Node,
  /** onClick handler passed down to the button */
  onClick?: Function,
  /** type of the button, 'default' | 'link' | 'add' | 'outline' */
  type?: ButtonType,
  /** state of the button, 'outline' | 'disabled' | 'loading' | 'icon' */
  state?: ButtonState,
  /** the color of the button, | 'primary' | 'secondary' | 'warning' | 'error' | 'success' */
  colorVariant?: ButtonColorVariant,
  /** the element to be placed on the button */
  icon?: React.Node,
  /** if the button is in disabled state */
  disabled?: ButtonDisabled,
  /** text to be rendered on the loading state */
  loadingText?: React.Node,
  /** text to be rendered on the error state */
  errorText?: React.Node,
  /** text to be rendered on the success state */
  successText?: React.Node,
};

const Icon = styled.span`
  margin-right: 10px;
  height: 18px;
`;

const textToRender = (
  state?: ButtonState,
  children: React.Node,
  loadingText: React.Node,
  errorText: React.Node,
  successText: React.Node,
) => {
  switch (state) {
    case 'loading':
      return loadingText;
    case 'error':
      return errorText;
    case 'success':
      return successText;
    default:
      return children;
  }
};

const Button = ({
  children,
  type,
  state,
  icon,
  colorVariant,
  disabled,
  defaultText,
  loadingText,
  errorText,
  successText,
  ...rest
}: Props) => (
  <SButton
    options={{ type, state, colorVariant, disabled }}
    disabled={
      state === 'loading' ||
      state === 'error' ||
      state === 'success' ||
      disabled
    }
    {...rest}
  >
    {icon ? <Icon>{icon}</Icon> : null}{' '}
    {state === 'loading' ? (
      <CircularProgress
        style={{
          height: '18px',
          width: '18px',
          marginRight: '10px',
          color:
            type === 'add'
              ? colors.secondary.loading
              : type === 'outline'
                ? colors[colorVariant].default
                : colors.gray.loading,
        }}
      />
    ) : null}{' '}
    <span>
      {textToRender(state, children, loadingText, errorText, successText)}
    </span>
  </SButton>
);

Button.defaultProps = {
  onClick: () => {},
  type: 'default',
  state: 'default',
  icon: null,
  disabled: false,
  colorVariant: 'primary',
  loadingText: '',
  errorText: '',
  successText: '',
};

export default Button;
