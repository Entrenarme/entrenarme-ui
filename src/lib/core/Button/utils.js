// @flow
import * as React from 'react';

import colors from '../../helpers/colors';

export const getButtonHeight = (size: ButtonSize, device: Device) => {
  switch (size) {
    case 'small':
      if (device === 'mobile' || device === 'iphone') return '30px';
      return '34px';
    case 'large':
      if (device === 'mobile' || device === 'iphone') return '40px';
      return '46px';
    case 'xlarge':
      if (device === 'mobile' || device === 'iphone') return '46px';
      return '54px';
    default:
      if (device === 'mobile' || device === 'iphone') return '36px';
      return '40px';
  }
};

export const getButtonPadding = (size: ButtonSize, device: Device) => {
  switch (size) {
    case 'small':
      if (device === 'mobile' || device === 'iphone')
        return '0 12px !important';
      return '0 15px !important';
    case 'large':
      if (device === 'mobile' || device === 'iphone')
        return '0 20px !important';
      return '0 20px !important';
    case 'xlarge':
      if (device === 'mobile' || device === 'iphone')
        return '0 20px !important';
      return '0 24px !important';
    default:
      if (device === 'mobile' || device === 'iphone')
        return '0 16px !important';
      return '0 20px !important';
  }
};

export const getButtonFontSize = (size: ButtonSize, device: Device) => {
  switch (size) {
    case 'small':
      if (device === 'mobile' || device === 'iphone') return '0.625rem';
      return '0.75rem';
    case 'large':
      if (device === 'mobile' || device === 'iphone') return '0.875rem';
      return '1rem';
    case 'xlarge':
      if (device === 'mobile' || device === 'iphone') return '1rem';
      return '1.125rem';
    default:
      if (device === 'mobile' || device === 'iphone') return '0.75rem';
      return '0.875rem';
  }
};

export const getButtonLineHeight = (size: ButtonSize, device: Device) => {
  switch (size) {
    case 'small':
      if (device === 'mobile' || device === 'iphone') return '10px';
      return '10px';
    case 'large':
      if (device === 'mobile' || device === 'iphone') return '10px';
      return '14px';
    case 'xlarge':
      if (device === 'mobile' || device === 'iphone') return '14px';
      return '14px';
    default:
      if (device === 'mobile' || device === 'iphone') return '12px';
      return '10px';
  }
};

export const textToRender = (
  state?: ButtonState,
  children: React.Node,
  loadingText: React.Node,
  errorText: React.Node,
  successText: React.Node,
) => {
  switch (state) {
    case 'loading':
      return loadingText;
    case 'error':
      return errorText;
    case 'success':
      return successText;
    default:
      return children;
  }
};

export const getBgColor = (
  buttonType: ButtonType,
  options: 'default' | 'active',
  state: ButtonState,
  disabled: ButtonDisabled,
  colorVariant: ButtonColorVariant,
  active: boolean,
) => {
  if (buttonType === 'link' || buttonType === 'outline') {
    return colors.white[active ? 'active' : 'default'];
  }
  if (buttonType === 'add' || disabled) {
    return colors.gray[active ? 'active' : 'default'];
  }
  if (state === 'error' || state === 'success') {
    return colors[state][active ? 'active' : 'default'];
  }
  return colors[colorVariant || state][active ? 'active' : 'default'];
};

export const getColor = (
  buttonType: ButtonType,
  state: ButtonState,
  colorVariant: ButtonColorVariant,
  disabled: ButtonDisabled,
) => {
  if (buttonType === 'outline') {
    return colors[colorVariant].default;
  }
  if (disabled) {
    return colors.gray.dark;
  }
  if (state === 'loading') {
    if (buttonType === 'link') {
      return colors.gray.loading;
    }
    if (buttonType === 'add') {
      return colors.secondary.loading;
    }
    return colors.white.loading;
  }
  if (buttonType === 'link' || buttonType === 'add') {
    return colors.secondary.default;
  }
  return colors.white.default;
};
