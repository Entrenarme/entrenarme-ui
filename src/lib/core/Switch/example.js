export default `Switch

## Basic usage
<Switch /> renders a function and give this props:
  - itemProps
    - A function when executed, it returns an object with useful props for each item.
    - It has to be applied to each item we want to include in our selection (example below).
  - activeItem
    - A string or Array<string> which will contain the selected name property of the active items,
      so you can check which one is active.

~~~js
  <Switch onChange={(activeItems) => console.log('will print male or female', activeItems)}>
  {({ itemProps, activeItem }) => (
    <React.Fragment>
      <Button
        name="male"
        type="outline"
        colorVariant={activeItem === 'male' ? 'secondary' : 'primary'}
        {...itemProps()}
      >
        no icon
      </Button>
      <Button
        name="female"
        type="outline"
        icon={<FontAwesomeIcon icon={faCoffee} />}
        colorVariant={activeItem === 'female' ? 'secondary' : 'primary'}
        {...itemProps()}
      >
        with icon
      </Button>
    </React.Fragment>
  )}
  </Switch>
~~~

## Multiple usage
Same as before, but with the prop multiple passed to <Switch /> component, as the example below.

~~~js
  <Switch onChange={(activeItems) => console.log('will print an array of ACTIVE strings ['male', 'female'], activeItems)} multiple>
  {({ itemProps, activeItem }) => (
    <React.Fragment>
      <Button
        name="male"
        type="outline"
        colorVariant={activeItem === 'male' ? 'secondary' : 'primary'}
        {...itemProps()}
      >
        no icon
      </Button>
      <Button
        name="female"
        type="outline"
        icon={<FontAwesomeIcon icon={faCoffee} />}
        colorVariant={activeItem === 'female' ? 'secondary' : 'primary'}
        {...itemProps()}
      >
        with icon
      </Button>
    </React.Fragment>
  )}
  </Switch>
~~~

`;
