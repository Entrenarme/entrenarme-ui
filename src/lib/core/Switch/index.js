// @flow
import * as React from 'react';
import styled from 'styled-components';

import Select from '../Select';

type ContainerProps = {
  options: {
    column: boolean,
  },
};

const Container = styled.div`
  display: grid;
  grid-template-columns: ${(props: ContainerProps) =>
    props.options.column ? '1fr' : '1fr 1fr'};
  grid-gap: 20px;
`;

type Props = {
  children: Function,
  column?: boolean,
};

const Switch = ({ children, column, ...rest }: Props) => (
  <Container options={{ column }}>
    <Select {...rest}>{children}</Select>
  </Container>
);

Switch.defaultProps = {
  column: false,
};

export default Switch;
