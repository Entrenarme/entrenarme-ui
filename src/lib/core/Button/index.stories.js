// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCoffee } from '@fortawesome/free-solid-svg-icons';

import Button from './index';

const actions = {
  onClick: action('onClick'),
};

const createButton = (
  {
    type = 'primary',
    state,
    colorVariant,
    disabled,
    icon,
    size,
    active,
  }: {
    type?: ButtonType,
    state?: ButtonState,
    colorVariant?: ButtonColorVariant,
    disabled?: ButtonDisabled,
    icon?: React.Node,
    size?: ButtonSize,
    active?: boolean,
  } = { colorVariant: 'primary' },
) => (
  <Button
    type={type}
    state={state}
    colorVariant={colorVariant}
    disabled={disabled}
    icon={icon}
    size={size}
    loadingText="loading"
    errorText="error"
    successText="success"
    active={active}
    {...actions}
  >
    button {colorVariant}
  </Button>
);

storiesOf('Buttons/Default', module)
  .add('default (primary)', () => createButton())
  .add('default active (primary)', () => createButton({ active: true }))
  .add('default small (primary)', () => createButton({ size: 'small' }))
  .add('default large (primary)', () => createButton({ size: 'large' }))
  .add('default xlarge (primary)', () => createButton({ size: 'xlarge' }))
  .add('default fullwidth (primary)', () => createButton({ size: 'fullwidth' }))
  .add('secondary', () => createButton({ colorVariant: 'secondary' }))
  .add('warning', () => createButton({ colorVariant: 'warning' }))
  .add('loading', () => createButton({ state: 'loading' }))
  .add('error color', () => createButton({ colorVariant: 'error' }))
  .add('error state', () => createButton({ state: 'error' }))
  .add('success color', () => createButton({ colorVariant: 'success' }))
  .add('success state', () => createButton({ state: 'success' }))
  .add('icon', () =>
    createButton({
      icon: <FontAwesomeIcon icon={faCoffee} />,
    }),
  );

storiesOf('Buttons/Outline', module)
  .add('default (primary)', () =>
    createButton({ colorVariant: 'primary', type: 'outline' }),
  )
  .add('secondary', () =>
    createButton({ colorVariant: 'secondary', type: 'outline' }),
  )
  .add('warning', () =>
    createButton({ colorVariant: 'warning', type: 'outline' }),
  )
  .add('loading', () => createButton({ type: 'outline', state: 'loading' }))
  .add('error', () => createButton({ type: 'outline', colorVariant: 'error' }))
  .add('success', () =>
    createButton({ type: 'outline', colorVariant: 'success' }),
  )
  .add('icon', () =>
    createButton({
      type: 'outline',
      icon: <FontAwesomeIcon icon={faCoffee} />,
    }),
  );

storiesOf('Buttons/Link', module)
  .add('default', () => createButton({ type: 'link' }))
  .add('disabled', () => createButton({ type: 'link', disabled: true }))
  .add('loading', () => createButton({ type: 'link', state: 'loading' }))
  .add('icon', () =>
    createButton({
      type: 'link',
      icon: <FontAwesomeIcon icon={faCoffee} />,
    }),
  );

storiesOf('Buttons/Add', module)
  .add('default', () => createButton({ type: 'add' }))
  .add('disabled', () => createButton({ type: 'add', disabled: true }))
  .add('loading', () => createButton({ state: 'loading', type: 'add' }))
  .add('icon', () =>
    createButton({
      type: 'add',
      icon: <FontAwesomeIcon icon={faCoffee} />,
    }),
  );
