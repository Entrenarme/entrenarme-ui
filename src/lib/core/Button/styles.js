// @flow
import styled from 'styled-components';
import MButton from '@material-ui/core/Button';

import * as fonts from '../../helpers/fonts';
import colors from '../../helpers/colors';

import { getBgColor, getColor } from './utils';

type ButtonProps = {
  options: {
    type: ButtonType,
    state: ButtonState,
    colorVariant: ButtonColorVariant,
    disabled: ButtonDisabled,
  },
};

export const SButton = styled(MButton)`
  height: 40px;
  box-sizing: border-box;
  background-color: ${(props: ButtonProps) =>
    getBgColor(
      props.options.type,
      'default',
      props.options.state,
      props.options.disabled,
      props.options.colorVariant,
    )} !important;
  ${(props: ButtonProps) =>
    props.options.type === 'outline'
      ? `	border: 2px solid ${getColor(
          props.options.type,
          props.options.state,
          props.options.colorVariant,
          props.options.disabled,
        )} !important`
      : props.options.type === 'add'
        ? `border: 2px dashed ${colors.gray.dark} !important;`
        : ''};
  :hover {
    background-color: ${(props: ButtonProps) =>
      getBgColor(
        props.options.type,
        'active',
        props.options.state,
        props.options.disabled,
        props.options.colorVariant,
      )} !important;
  }
  span {
    color: ${(props: ButtonProps) =>
      getColor(
        props.options.type,
        props.options.state,
        props.options.colorVariant,
        props.options.disabled,
      )};
    font-family: ${fonts.mainFont};
  }
`;

export const Icon = styled.span`
  margin-right: 10px;
  height: 18px;
`;
