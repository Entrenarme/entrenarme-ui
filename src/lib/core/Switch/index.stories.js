// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withInfo } from '@storybook/addon-info';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCoffee } from '@fortawesome/free-solid-svg-icons';

import Switch from './index';
import Button from '../Button';

storiesOf('Switch', module).add(
  'default',
  withInfo('Switch')(() => (
    <Switch onChange={action('onSwitchChange')}>
      {({ itemProps, activeItem }) => (
        <React.Fragment>
          <Button
            name="male"
            type="outline"
            colorVariant={activeItem === 'male' ? 'secondary' : 'primary'}
            {...itemProps()}
          >
            no icon
          </Button>
          <Button
            name="female"
            type="outline"
            icon={<FontAwesomeIcon icon={faCoffee} />}
            colorVariant={activeItem === 'female' ? 'secondary' : 'primary'}
            {...itemProps()}
          >
            with icon
          </Button>
        </React.Fragment>
      )}
    </Switch>
  )),
);
