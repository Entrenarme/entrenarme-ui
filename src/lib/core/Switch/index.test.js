import * as React from 'react';
import { render, fireEvent } from 'react-testing-library';

import Switch from './index';
import Button from '../Button';

test('render props are correct with switch', () => {
  let activeItems = null;
  const onChange = jest.fn();
  const { queryAllByText, container } = render(
    <Switch onChange={onChange}>
      {({ itemProps, activeItem }) => {
        activeItems = activeItem;
        return (
          <React.Fragment>
            <Button
              name="male"
              type="outline"
              colorVariant={activeItem === 'male' ? 'secondary' : 'primary'}
              {...itemProps()}
            >
              no icon
            </Button>
            <Button
              name="female"
              type="outline"
              colorVariant={activeItem === 'female' ? 'secondary' : 'primary'}
              {...itemProps()}
            >
              with icon
            </Button>
          </React.Fragment>
        );
      }}
    </Switch>,
  );

  const noIconTextButton = queryAllByText('no icon');
  const withIconTextButton = queryAllByText('with icon');

  expect(noIconTextButton).toHaveLength(1);
  expect(withIconTextButton).toHaveLength(1);

  expect(activeItems).toBeFalsy();

  expect(container).toMatchSnapshot('none active');

  fireEvent.click(noIconTextButton[0]);
  expect(activeItems).toBe('male');
  expect(onChange).toHaveBeenCalledTimes(1);
  expect(onChange).toHaveBeenLastCalledWith('male');

  expect(container).toMatchSnapshot('first active');

  fireEvent.click(noIconTextButton[0]);
  expect(activeItems).toBeFalsy();
  expect(onChange).toHaveBeenCalledTimes(2);
  expect(onChange).toHaveBeenLastCalledWith(undefined);

  expect(container).toMatchSnapshot('none active');

  fireEvent.click(noIconTextButton[0]);
  expect(activeItems).toBe('male');
  expect(onChange).toHaveBeenCalledTimes(3);
  expect(onChange).toHaveBeenLastCalledWith('male');

  expect(container).toMatchSnapshot('first active');

  fireEvent.click(withIconTextButton[0]);
  expect(activeItems).toBe('female');
  expect(onChange).toHaveBeenCalledTimes(4);
  expect(onChange).toHaveBeenLastCalledWith('female');

  expect(container).toMatchSnapshot('second active');

  fireEvent.click(withIconTextButton[0]);
  expect(activeItems).toBeFalsy();
  expect(onChange).toHaveBeenCalledTimes(5);
  expect(onChange).toHaveBeenLastCalledWith(undefined);

  expect(container).toMatchSnapshot('none active');
});

test('render props are correct with multiple', () => {
  let activeItems = null;
  const onChange = jest.fn();
  const { queryAllByText, container } = render(
    <Switch onChange={onChange} multiple>
      {({ itemProps, activeItem }) => {
        activeItems = activeItem;
        return (
          <React.Fragment>
            <Button
              name="male"
              type="outline"
              colorVariant={activeItem === 'male' ? 'secondary' : 'primary'}
              {...itemProps()}
            >
              no icon
            </Button>
            <Button
              name="female"
              type="outline"
              colorVariant={activeItem === 'female' ? 'secondary' : 'primary'}
              {...itemProps()}
            >
              with icon
            </Button>
          </React.Fragment>
        );
      }}
    </Switch>,
  );

  const noIconTextButton = queryAllByText('no icon');
  const withIconTextButton = queryAllByText('with icon');

  expect(noIconTextButton).toHaveLength(1);
  expect(withIconTextButton).toHaveLength(1);

  expect(activeItems).toEqual([]);

  expect(container).toMatchSnapshot('none active');

  fireEvent.click(noIconTextButton[0]);
  expect(activeItems).toEqual(['male']);
  expect(onChange).toHaveBeenCalledTimes(1);
  expect(onChange).toHaveBeenLastCalledWith(['male']);

  expect(container).toMatchSnapshot('first active');

  fireEvent.click(noIconTextButton[0]);
  expect(activeItems).toEqual([]);
  expect(onChange).toHaveBeenCalledTimes(2);
  expect(onChange).toHaveBeenLastCalledWith([]);

  expect(container).toMatchSnapshot('none active');

  fireEvent.click(noIconTextButton[0]);
  expect(activeItems).toEqual(['male']);
  expect(onChange).toHaveBeenCalledTimes(3);
  expect(onChange).toHaveBeenLastCalledWith(['male']);

  expect(container).toMatchSnapshot('first active');

  fireEvent.click(withIconTextButton[0]);
  expect(activeItems).toEqual(['male', 'female']);
  expect(onChange).toHaveBeenCalledTimes(4);
  expect(onChange).toHaveBeenLastCalledWith(['male', 'female']);

  expect(container).toMatchSnapshot('both active');

  fireEvent.click(withIconTextButton[0]);
  expect(activeItems).toEqual(['male']);
  expect(onChange).toHaveBeenCalledTimes(5);
  expect(onChange).toHaveBeenLastCalledWith(['male']);

  expect(container).toMatchSnapshot('first active');
});
