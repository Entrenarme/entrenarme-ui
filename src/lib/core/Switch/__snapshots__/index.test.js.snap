// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`both active 1`] = `
<div>
  <div
    class="sc-bdVaJa bBzjfv"
  >
    <button
      tabindex="0"
      class="MuiButtonBase-root-59 MuiButton-root-35 MuiButton-text-37 MuiButton-flat-40 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="male"
    >
      <span
        class="MuiButton-label-36"
      >
        <span>
          no icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-62"
      />
    </button>
    <button
      tabindex="0"
      class="MuiButtonBase-root-59 MuiButton-root-35 MuiButton-text-37 MuiButton-flat-40 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="female"
    >
      <span
        class="MuiButton-label-36"
      >
        <span>
          with icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-62"
      />
    </button>
  </div>
</div>
`;

exports[`first active 1`] = `
<div>
  <div
    class="sc-bdVaJa bBzjfv"
  >
    <button
      tabindex="0"
      class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bwzfXH jBBLTz"
      type="button"
      options="[object Object]"
      name="male"
    >
      <span
        class="MuiButton-label-2"
      >
        <span>
          no icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-28"
      />
    </button>
    <button
      tabindex="0"
      class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="female"
    >
      <span
        class="MuiButton-label-2"
      >
        <span>
          with icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-28"
      />
    </button>
  </div>
</div>
`;

exports[`first active 2`] = `
<div>
  <div
    class="sc-bdVaJa bBzjfv"
  >
    <button
      tabindex="0"
      class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bwzfXH jBBLTz"
      type="button"
      options="[object Object]"
      name="male"
    >
      <span
        class="MuiButton-label-2"
      >
        <span>
          no icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-28"
      />
    </button>
    <button
      tabindex="0"
      class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="female"
    >
      <span
        class="MuiButton-label-2"
      >
        <span>
          with icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-28"
      />
    </button>
  </div>
</div>
`;

exports[`first active 3`] = `
<div>
  <div
    class="sc-bdVaJa bBzjfv"
  >
    <button
      tabindex="0"
      class="MuiButtonBase-root-59 MuiButton-root-35 MuiButton-text-37 MuiButton-flat-40 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="male"
    >
      <span
        class="MuiButton-label-36"
      >
        <span>
          no icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-62"
      />
    </button>
    <button
      tabindex="0"
      class="MuiButtonBase-root-59 MuiButton-root-35 MuiButton-text-37 MuiButton-flat-40 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="female"
    >
      <span
        class="MuiButton-label-36"
      >
        <span>
          with icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-62"
      />
    </button>
  </div>
</div>
`;

exports[`first active 4`] = `
<div>
  <div
    class="sc-bdVaJa bBzjfv"
  >
    <button
      tabindex="0"
      class="MuiButtonBase-root-59 MuiButton-root-35 MuiButton-text-37 MuiButton-flat-40 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="male"
    >
      <span
        class="MuiButton-label-36"
      >
        <span>
          no icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-62"
      />
    </button>
    <button
      tabindex="0"
      class="MuiButtonBase-root-59 MuiButton-root-35 MuiButton-text-37 MuiButton-flat-40 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="female"
    >
      <span
        class="MuiButton-label-36"
      >
        <span>
          with icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-62"
      />
    </button>
  </div>
</div>
`;

exports[`first active 5`] = `
<div>
  <div
    class="sc-bdVaJa bBzjfv"
  >
    <button
      tabindex="0"
      class="MuiButtonBase-root-59 MuiButton-root-35 MuiButton-text-37 MuiButton-flat-40 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="male"
    >
      <span
        class="MuiButton-label-36"
      >
        <span>
          no icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-62"
      />
    </button>
    <button
      tabindex="0"
      class="MuiButtonBase-root-59 MuiButton-root-35 MuiButton-text-37 MuiButton-flat-40 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="female"
    >
      <span
        class="MuiButton-label-36"
      >
        <span>
          with icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-62"
      />
    </button>
  </div>
</div>
`;

exports[`none active 1`] = `
<div>
  <div
    class="sc-bdVaJa bBzjfv"
  >
    <button
      tabindex="0"
      class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="male"
    >
      <span
        class="MuiButton-label-2"
      >
        <span>
          no icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-28"
      />
    </button>
    <button
      tabindex="0"
      class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="female"
    >
      <span
        class="MuiButton-label-2"
      >
        <span>
          with icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-28"
      />
    </button>
  </div>
</div>
`;

exports[`none active 2`] = `
<div>
  <div
    class="sc-bdVaJa bBzjfv"
  >
    <button
      tabindex="0"
      class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="male"
    >
      <span
        class="MuiButton-label-2"
      >
        <span>
          no icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-28"
      />
    </button>
    <button
      tabindex="0"
      class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="female"
    >
      <span
        class="MuiButton-label-2"
      >
        <span>
          with icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-28"
      />
    </button>
  </div>
</div>
`;

exports[`none active 3`] = `
<div>
  <div
    class="sc-bdVaJa bBzjfv"
  >
    <button
      tabindex="0"
      class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="male"
    >
      <span
        class="MuiButton-label-2"
      >
        <span>
          no icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-28"
      />
    </button>
    <button
      tabindex="0"
      class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="female"
    >
      <span
        class="MuiButton-label-2"
      >
        <span>
          with icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-28"
      />
    </button>
  </div>
</div>
`;

exports[`none active 4`] = `
<div>
  <div
    class="sc-bdVaJa bBzjfv"
  >
    <button
      tabindex="0"
      class="MuiButtonBase-root-59 MuiButton-root-35 MuiButton-text-37 MuiButton-flat-40 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="male"
    >
      <span
        class="MuiButton-label-36"
      >
        <span>
          no icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-62"
      />
    </button>
    <button
      tabindex="0"
      class="MuiButtonBase-root-59 MuiButton-root-35 MuiButton-text-37 MuiButton-flat-40 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="female"
    >
      <span
        class="MuiButton-label-36"
      >
        <span>
          with icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-62"
      />
    </button>
  </div>
</div>
`;

exports[`none active 5`] = `
<div>
  <div
    class="sc-bdVaJa bBzjfv"
  >
    <button
      tabindex="0"
      class="MuiButtonBase-root-59 MuiButton-root-35 MuiButton-text-37 MuiButton-flat-40 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="male"
    >
      <span
        class="MuiButton-label-36"
      >
        <span>
          no icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-62"
      />
    </button>
    <button
      tabindex="0"
      class="MuiButtonBase-root-59 MuiButton-root-35 MuiButton-text-37 MuiButton-flat-40 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="female"
    >
      <span
        class="MuiButton-label-36"
      >
        <span>
          with icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-62"
      />
    </button>
  </div>
</div>
`;

exports[`second active 1`] = `
<div>
  <div
    class="sc-bdVaJa bBzjfv"
  >
    <button
      tabindex="0"
      class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bwzfXH gjHic"
      type="button"
      options="[object Object]"
      name="male"
    >
      <span
        class="MuiButton-label-2"
      >
        <span>
          no icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-28"
      />
    </button>
    <button
      tabindex="0"
      class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bwzfXH jBBLTz"
      type="button"
      options="[object Object]"
      name="female"
    >
      <span
        class="MuiButton-label-2"
      >
        <span>
          with icon
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-28"
      />
    </button>
  </div>
</div>
`;
